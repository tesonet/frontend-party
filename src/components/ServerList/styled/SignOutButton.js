import React from 'react';
import styled from 'styled-components';
import SignOutIcon from 'react-icons/lib/go/sign-out';

const StyledIcon = styled(SignOutIcon)`
  height: auto;
  padding-left: .5rem;
  transform: rotateY(180deg);
  width: 1.5rem;
`;

const StyledContainer = styled.div`
  &:hover,
  &:focus {
    cursor: pointer;
  }
`;

const SignOutButton = props => (
  <StyledContainer {...props}>
    <StyledIcon />
    Logout
  </StyledContainer>
);

export default SignOutButton;
