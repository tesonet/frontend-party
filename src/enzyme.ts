import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
