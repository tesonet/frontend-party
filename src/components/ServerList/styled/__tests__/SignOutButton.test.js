import React from 'react';
import { shallow } from 'enzyme';

import SignOutButton from '../SignOutButton';

describe('SignOutButton component', () => {
  it('should render the <div /> element.', () => {
    const renderedComponent = shallow(<SignOutButton />).dive();

    expect(renderedComponent.type()).toEqual('div');
    expect(renderedComponent).toMatchSnapshot();
  });
});
