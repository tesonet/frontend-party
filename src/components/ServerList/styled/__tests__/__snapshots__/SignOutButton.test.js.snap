// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOutButton component should render the <div /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div>
    <Styled(GoSignOut) />
    Logout
</styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(GoSignOut) />,
        "Logout",
      ],
      "className": "sc-bwzfXH hGFsSt",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Logout",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(GoSignOut) />,
          "Logout",
        ],
        "className": "sc-bwzfXH hGFsSt",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Logout",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
